### ISSUE TITLE
Villager trades overpopulating

### ISSUE THREAD

**Issue by Dornar** [Globox1997/VillagerTradeFix#4]
### Minecraft Version

1.18.x

### Mod Loader Version

fabric 0.14.24

### Mod Version

1.0.2

### Describe the Issue

Testing a villager bug in server side, player showed a villager that they shuffled. The villagers trade table included shuffle A, then included A & B on second shuffle.

In my effort to test and recreate the issue, i found the bug preasent in sigle player as well. The villagers trade table includes trades [A], on first shuffle. Then no loss of trades A on shuffle two, rather includes trades [A, B]... same is true on shuffle three and so on... 
![2023-11-26_00 39 14](https://github.com/Globox1997/VillagerTradeFix/assets/88221028/ff8c7550-12ff-4e5a-a14a-dde5f93137b4)
![2023-11-26_00 39 30](https://github.com/Globox1997/VillagerTradeFix/assets/88221028/1a4e8f78-9b76-4de4-8517-90f038ca00b7)

Basically, when the player shuffles the tables. it is not stopping the new tables from being generated, but keeps the old tables in the traders storage.

### latest.log

[latest.log](https://github.com/Globox1997/VillagerTradeFix/files/13466080/latest.log)

### Do you use any other mods except the required ones?

when this mod is enabled, the bug shows. without this mod enabled. the bug is not present.
